File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
let input = fs.readFileSync(filePath).toString().trim().split("\n");
let nums = input.map((i) => i.split(" ").map((i) => +i));
// 누적합
let array = Array.from({ length: nums[1].length + 1 }, () => 0);
let answer = [];
nums[1].forEach((el, i) => {
  array[i + 1] = array[i] + el;
});
for (let i = 2; i < nums[0][1] + 2; i++) {
  let [start, end] = nums[i];
  answer.push(array[end] - array[start - 1]);
}
console.log(answer.join("\n"));

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.py
import sys
input = sys.stdin.readline
N, M = map(int, input().split())
numbers = list(map(int, input().split()))
sumNumbers = [0] # 합 배열 선언
temp = 0 # 임시 변수 선언

for i in numbers:
    temp += i
    sumNumbers.append(temp)

for k in range(M):
    i, j = map(int, input().split())
    print(sumNumbers[j] - sumNumbers[i - 1])